File: backend/internal/dto/response/currency_config.go
package response

type SupportedCurrenciesResp struct {
	CryptoCurrencies []string            `json:"crypto_currencies"`
	FiatCurrencies   []string            `json:"fiat_currencies"`
	CryptoChains     map[string][]string `json:"crypto_chains"`
	DefaultChains    map[string]string   `json:"default_chains"`
}

type SystemCurrencyConfigResp struct {
	CryptoCurrencies []string            `json:"crypto_currencies"`
	FiatCurrencies   []string            `json:"fiat_currencies"`
	CryptoChains     map[string][]string `json:"crypto_chains"`
	DefaultChains    map[string]string   `json:"default_chains"`
	CatalogChains    map[string][]string `json:"catalog_chains"`
	AllCrypto        []string            `json:"all_crypto"`
	AllFiat          []string            `json:"all_fiat"`
}

File: admin/src/types/currency-config.ts
export interface SystemCurrencyConfig {
  crypto_currencies: string[];
  fiat_currencies: string[];
  crypto_chains: Record<string, string[]>;
  default_chains: Record<string, string>;
  catalog_chains: Record<string, string[]>;
  all_crypto: string[];
  all_fiat: string[];
}

export interface UpdateSystemCurrencyConfigPayload {
  crypto_currencies: string[];
  fiat_currencies: string[];
  crypto_chains: Record<string, string[]>;
  default_chains: Record<string, string>;
}
